bin/analysis.cpp

```diff
@@ -237,6 +237,20 @@ int main(int ac, char* av[])
 		return 1;
 	}
 
+
+	std::cout << "[INFO] Selected modes: \n";
+	unsigned int msize = modes.size();
+	modes.erase( std::remove( modes.begin(), modes.end(), "signal" ), modes.end() );
+	if(modes.size() != msize) {
```

Comment:
When you say that modes are expanded before getChainNames you mean here, right?
Now I understand the logic of this. It seems to work, but it's very nasty. 
I'm wondering if we should define the modes in the datasets.dat as additional column, like:
datasetName ... ... ... mode
DYToEE        ... ... ... DY
and when you use "signal" as mode, then analysis.cpp looks into dataset.dat and make the push_back according to the samples defined there.
I would be nice if you can open an additional issue for this to be fixed in the next future.